getting_started/scripting/gdscript/static_typing.rst

```diff
@@ -0,0 +1,413 @@
+.. _doc_gdscript_types_intro:
+
+Static Typing in GDscript
+=========================
+
+In this guide, you will learn:
+
+-  **How to use types in GDscript**
+-  That **static types can help you avoid bugs**
+
+Where and how you use this new language feature is entirely up to you:
+you can use it only in some sensitive GDScript files, use it everywhere,
+or write code like you always did!
+
+Static types can be used on variables, constants, functions, parameters,
+and return types.
+
+.. note::
+
+    Typed GDscript is available since Godot 3.1.
+
+Why You Should Learn to Use Typed GDScript
+------------------------------------------
+
+With typed GDscript, Godot can detect even more errors as you write
+code! It gives you and your teammates more information as you’re
+working, as the arguments’ types show up when you call a method.
+
+Imagine you’re programming an inventory system. You code an ``Item``
+node, then an ``Inventory``. To add items to the inventory, the people
+who work with your code should always pass an ``Item`` to the
+``Inventory.add`` method. With types, you can force them to:
+
+::
+
+    """in Item.gd"""
+    class_name Item
+
+    """in Inventory.gd"""
+    class_name Inventory
+
+    func add(reference : Item, amount : int = 1):
+        var item = find_item(reference)
+        if not item:
+            item = _instance_item_from_db(reference)
+        item.amount += amount
+
+Another significant advantage of typed GDScript is the new **warning
+system**. From version 3.1, Godot gives you warnings about your code as
+you write it: the engine identifies sections of your code that may lead
+to issues at runtime, but lets you decide whether or not you want to
+leave the code as it is. More on that in a moment.
+
+Static types also give you better code completion options. Below, you
+can see the difference between a dynamic and a static typed completion
+options for a class called ``PlayerController``.
+
+You’ve probably stored a node in a variable before, and typed a dot to
+be left with no autocomplete suggestions:
+
+.. figure:: ./img/typed_gdscript_code_completion_dynamic.png
+   :alt: code completion options for dynamic
+
+This is due to dynamic code. Godot cannot know what node or value type
+you’re passing to the function. If you write the type explicitely
+however, you will get all public methods and variables from the node:
+
+.. figure:: ./img/typed_gdscript_code_completion_typed.png
+   :alt: code completion options for typed
+
+In the future, typed GDScript will also increase code performance: Just
+In Time compilation and other compiler improvements are already on the
+roadmap!
```

Comment:
Being a little nit-picky here, but: `Just-In-Time` (or even `JIT`, but I think I understand why you've written it out) reads a little better to me.